internal/plugins/helm/v1/scaffolds/templates/role.go

```diff
@@ -0,0 +1,502 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package templates
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/releaseutil"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/kubebuilder/pkg/model/file"
+	"sigs.k8s.io/kubebuilder/pkg/model/resource"
+	"sigs.k8s.io/yaml"
+)
+
+var _ file.Template = &Role{}
+
+// Role scaffolds the config/rbac/auth_proxy_role.yaml file
+type Role struct {
+	file.TemplateMixin
+
+	SkipDefaultRules bool
+	CustomRules      []rbacv1.PolicyRule
+}
+
+// SetTemplateDefaults implements input.Template
+func (f *Role) SetTemplateDefaults() error {
+	if f.Path == "" {
+		f.Path = filepath.Join("config", "rbac", "role.yaml")
+	}
+
+	f.TemplateBody = roleTemplate
+
+	return nil
+}
+
+const roleTemplate = `apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: manager-role
+rules:
+{{- if not .SkipDefaultRules }}
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  - services
+  - services/finalizers
+  - endpoints
+  - persistentvolumeclaims
+  - events
+  - configmaps
+  - secrets
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - apps
+  resources:
+  - deployments
+  - daemonsets
+  - replicasets
+  - statefulsets
+  verbs:
+  - create
+  - delete
+  - get
+  - list
+  - patch
+  - update
+  - watch
+{{- end }}
+{{- range .CustomRules }}
+- verbs:
+  {{- range .Verbs }}
+  - "{{ . }}"
+  {{- end }}
+  {{- with .APIGroups }}
+  apiGroups:
+  {{- range . }}
+  - "{{ . }}"
+  {{- end }}
+  {{- end }}
+  {{- with .Resources }}
+  resources:
+  {{- range . }}
+  - "{{ . }}"
+  {{- end }}
+  {{- end }}
+  {{- with .ResourceNames }}
+  resourceNames:
+  {{- range . }}
+  - "{{ . }}"
+  {{- end }}
+  {{- end }}
+  {{- with .NonResourceURLs }}
+  nonResourceURLs:
+  {{- range . }}
+  - "{{ . }}"
+  {{- end }}
+  {{- end }}
+{{- end }}
+- apiGroups:
+  - monitoring.coreos.com
+  resources:
+  - servicemonitors
+  verbs:
+  - "get"
+  - "create"
+- apiGroups:
+  - apps
+  resources:
+  - deployments/finalizers
+  resourceNames:
+  - ""
+  verbs:
+  - "update"
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - get
+- apiGroups:
+  - apps
+  resources:
+  - replicasets
+  - deployments
+  verbs:
+  - get
```

Comment:
These are currently required by the helm-operator to create the servicemonitor with a valid owner reference.

However we also scaffold a service monitor resource in `./config/prometheus/monitor.yaml`. IMO, we should remove the code from helm-operator the generates the service monitor as a follow-up to this PR and before 1.0.

Can you add a TODO comment and capture this in a new GH issue in the 1.0 milestone?

Reply:
Sure 👍

Reply:
https://github.com/operator-framework/operator-sdk/issues/3354